src/common/secret.c

```diff
@@ -107,7 +121,7 @@ int get_secret_option(const char *secret, const char *key_name,
   if (secret) {
     ret = set_kernel_secret(secret, key_name);
     if (ret < 0) {
-      if (ret == -ENODEV || ret == -ENOSYS) {
+      if (ret == -ENODEV || ret == -ENOSYS || ret == -EINVAL) {
```

Comment:
I'd rather we have exemptions in the build system, rather than ifdefs in the code, especially when the entire file is to be excluded.  secret.c is shared between krbd.cc and mount.ceph.c and there is no reason for anything else to include it.

Just like krbd.cc and mount.ceph.c, secret.c shouldn't be built on anything other than Linux.